src/main/java/com/github/reducktion/socrates/validator/GermanyIdValidator.java

```diff
@@ -0,0 +1,135 @@
+package com.github.reducktion.socrates.validator;
+
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * A national identification number does not really exists within Germany.
+ * But there's a tax ident number, which is unique for each citizen and assigned by date of birth.
+ * This one is restricted by law to be used as a marker/attribute for tax/finance relevant purposes only,
+ * and not to identify people in general.
+ * It's called 'steuerliche Identifikationsnummer' or abbreviated 'Steuer-IdNr.'.
+ * There's no meta data, which could be extracted from this number.
+ *
+ * See https://de.wikipedia.org/wiki/Steuerliche_Identifikationsnummer
+ * The algorithm is published in an official document
+ * from: Informations Technik Zentrum Bund
+ * title: Steueridentifikationsnummer (IdNr) nach § 139b AO; Informationen zur Berechnung gültiger Prüfziffern
+ */
+public class GermanyIdValidator implements IdValidator {
+
+    private static final int ID_NUMBER_OF_CHARACTERS = 11;
+    private static final int CHECKSUM_INDEX = 10;
+
+    @Override
+    public boolean validate(final String value) {
+        if (value == null) {
+            return false;
+        }
+        if (value.length() != ID_NUMBER_OF_CHARACTERS) {
+            return false;
+        }
+        if (containsNonDigits(value)) {
```

Comment:
You can use our method `StringUtils.isNumeric(value)`, for this purpose.

Reply:
Thanks for the hint. Will do.